fitbenchmarking/core/tests/test_fitting_benchmarking_hessians.py

```diff
@@ -0,0 +1,221 @@
+        """
+        self.options.hes_method = ["analytic"]
+        self.controller.has_jacobian = [True]
+        self.controller.invalid_jacobians = ["deriv_free_algorithm"]
+        self.controller.minimizer = "deriv_free_algorithm"
+        self.controller.has_hessian = [False]
+        self.controller.valid_hessians = ["general"]
+        new_name = ['deriv_free_algorithm']
+        jacobian = False
+        minimizer_name = "deriv_free_algorithm"
+        _, _, new_minimizer_list = \
+            loop_over_hessians(self.controller,
+                               self.options,
+                               minimizer_name,
+                               jacobian,
+                               self.grabbed_output)
+        assert new_minimizer_list == new_name
+
+    # pylint: disable=unused-argument
+    @patch.object(DummyController, "cleanup")
+    @patch.object(DummyController, "check_bounds_respected")
+    def test_bounds_respected_func_called(
+            self, check_bounds_respected, cleanup):
+        """
+        Test that the check to verify that bounds are respected is called when
+        the controller runs succesfully.
+        """
+        self.controller.problem.value_ranges = {'test': (0, 1)}
+        self.controller.has_jacobian = [True]
+        self.controller.invalid_jacobians = ["deriv_free_algorithm"]
+        self.controller.minimizer = "deriv_free_algorithm"
+        self.controller.has_hessian = [False]
+        self.controller.valid_hessians = ["general"]
+        jacobian = False
+        minimizer_name = "deriv_free_algorithm"
+
+        # Cleanup has been mocked out with a no-op, so set the outputs now.
+        self.controller.flag = 0
+        self.controller.final_params = [1, 2, 3, 4]
+
+        _ = loop_over_hessians(self.controller,
+                               self.options,
+                               minimizer_name,
+                               jacobian,
+                               self.grabbed_output)
+        check_bounds_respected.assert_called()
+
+    @patch.object(DummyController, "cleanup")
+    @patch.object(DummyController, "check_bounds_respected")
+    def test_bounds_respected_func_not_called(
+            self, check_bounds_respected, cleanup):
+        """
+        Test that the check to verify that bounds are respected is not called
+        when the controller fails.
+        """
+        self.controller.problem.value_ranges = {'test': (0, 1)}
+        self.controller.has_jacobian = [True]
+        self.controller.invalid_jacobians = ["deriv_free_algorithm"]
+        self.controller.minimizer = "deriv_free_algorithm"
+        self.controller.has_hessian = [False]
+        self.controller.valid_hessians = ["general"]
+        jacobian = False
+        minimizer_name = "deriv_free_algorithm"
+
+        # Cleanup has been mocked out with a no-op, so set the outputs now.
+        self.controller.flag = 3
+        self.controller.final_params = [1, 2, 3, 4]
```

Comment:
I'm a bit confused at why this is needed, it looks like we can just set "final_params_expected" and "flag_expected" like you do with has_hessian

Reply:
Ahh, I've just spotted this is how it's done in jacobians... we can save this for when #901 is looked at